nemo/collections/asr/parts/utils/nmesc_clustering.py

```diff
@@ -1253,3 +1787,612 @@ def forward_infer(
         spectral_model = SpectralClustering(n_clusters=n_clusters, cuda=self.cuda, device=self.device)
         Y = spectral_model.forward(affinity_mat)
         return Y
+
+
+class OnlineSpeakerClustering:
```

Comment:
This file (`utils/nmesc_clustering.py`) is massive, should we create a folder within this general NeMo directory and split things out? i.e

`nemo/collections/asr/parts/utils/diarization/online_clustering.py`

Reply:
changed nmesc_clustering.py to offline_clustering.py then added online_clustering.py for all the newly added online clustering modules